zebra-network/src/peer_set/initialize.rs

```diff
@@ -359,26 +361,38 @@ where
 /// `peerset_initial_target_size`.
 ///
 /// The result is randomly chosen entries from the provided set of addresses.
-async fn limit_initial_peers(config: &Config) -> HashSet<SocketAddr> {
-    let initial_peers = config.initial_peers().await;
-    let initial_peer_count = initial_peers.len();
+async fn limit_initial_peers(
+    config: &Config,
+    mut address_book_updater: mpsc::Sender<MetaAddrChange>,
+) -> HashSet<SocketAddr> {
+    let all_peers = config.initial_peers().await;
+    let peers_count = all_peers.len();
+
+    if peers_count <= config.peerset_initial_target_size {
+        return all_peers;
+    }
 
     // Limit the number of initial peers to `config.peerset_initial_target_size`
-    if initial_peer_count > config.peerset_initial_target_size {
-        info!(
-            "Limiting the initial peers list from {} to {}",
-            initial_peer_count, config.peerset_initial_target_size
-        );
-    }
+    info!(
+        "Limiting the initial peers list from {} to {}",
+        peers_count, config.peerset_initial_target_size
+    );
 
-    let initial_peers_vect: Vec<SocketAddr> = initial_peers.iter().copied().collect();
+    // Split all the peers into the `initial_peers` that will be returned and
+    // `unused_peers` that will be sent to the address book.
+    let mut all_peers: Vec<SocketAddr> = all_peers.iter().copied().collect();
```

Comment:
Minor improvement: I think this could be moved instead of copied:

```suggestion
    let mut all_peers: Vec<SocketAddr> = all_peers.into_iter().collect();
```

Reply:
@jvff is it because we're not using the old `all_peers` anymore? I thought I'd ask so that I don't make the same mistake anymore.

Reply:
Yes, exactly. `.into_iter()` moves the collection and its items into the iterator, and `.collect()` moves the items from the iterator into the new collection. `.iter()` borrows the collection and the items into the iterator, `.copied()` copies the items so that they aren't borrows (i.e., `&T -> T`, as long as `T: Copy`, otherwise `.cloned()` can be used for `T: Clone`), and `.collect()` moves those copies into the new collection.

Since we're throwing away the old collection after its items are copied, its items are also dropped, so a mini-optimization is to move them instead.

However, I wouldn't be surprised if LLVM is smart enough to detect it's copying data then dropping the source, and convert it to moves anyway. So this sort of change often makes more sense to improve readability, conciseness and predictability than actually improving performance :sweat_smile: It would make sense for performance if it was `.cloned()` and `T::clone()` was expensive.